juju/model.py

```diff
@@ -257,7 +259,9 @@ def get_entity(self, entity_type, entity_id, history_index=-1, connected=True):
 class ModelEntity:
     """An object in the Model tree"""
 
-    def __init__(self, entity_id, model, history_index=-1, connected=True):
+    entity_id: str
+
+    def __init__(self, entity_id: str, model: Model, history_index=-1, connected=True):
```

Comment:
pyright seems to imply the type based on the default.
is that not enough?

Reply:
My concern is that a later reader doesn't necessarily know whether they're not type annotated because the type inferred from the defaults is obviously correct, or because it's like `wait_for_idle` where we're just gradually adding types. Whereas if you add even the obvious annotations, it's clear that the types for this `__init__` method have been fully worked out.

Reply:
noted.